File: AD-HOC/1105/main.py
def is_possible(b, n):
    banks = list(map(int, input().split()))
    for _ in range(n):
        d, c, v = map(int, input().split())
        banks[d-1] -= v
        banks[c-1] += v
    for reserve in banks:
        if reserve < 0:
            return False
    return True


b, n = map(int, input().split())
while (b or n):
    print('S' if is_possible(b, n) else 'N')
    b, n = map(int, input().split())

File: AD-HOC/1105/main.rb
def is_possible(b, n)
  banks = gets.split.map(&:to_i)
  n.times do
    d, c, v = gets.split.map(&:to_i)
    banks[d - 1] -= v
    banks[c - 1] += v
  end
  banks.each do |reserve|
    return false if reserve.negative?
  end
  true
end

b, n = gets.split.map(&:to_i)
while b.nonzero? || n.nonzero?
  puts is_possible(b, n) ? "S" : "N"
  b, n = gets.split.map(&:to_i)
end
